rama-http/src/headers/x_robots_tag_components/max_image_preview_setting.rs

```diff
@@ -0,0 +1,38 @@
+use rama_core::error::OpaqueError;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(super) enum MaxImagePreviewSetting {
+    None,
+    Standard,
+    Large,
+}
+
+impl std::fmt::Display for MaxImagePreviewSetting {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MaxImagePreviewSetting::None => write!(f, "none"),
```

Comment:
personally I would either turn these strings into constants or do the entire thing with a single custom private macro. Both approaches result in the same, that you only have to write these magic strings once.